### PR TITLE
GPU instancing in Lightweight renderpipeline

### PR DESCRIPTION
**Pull request by Wadarass**

### CODE REVIEW

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl [hunk 1]
```diff
@@ -221,10 +216,11 @@
     #endif
 
     #if defined(UNITY_INSTANCED_SH) && !defined(LIGHTMAP_ON)
-        #if UNITY_SHOULD_SAMPLE_SH
+        //In HDRenderpipe we only decide to look at probe data based on Lightmap flags.
+        #if !defined(DYNAMICLIGHTMAP_ON) 
             #define UNITY_USE_SHCOEFFS_ARRAYS
         #endif
-        #if defined(UNITY_PASS_DEFERRED) && defined(SHADOWS_SHADOWMASK) && (UNITY_ALLOWED_MRT_COUNT > 4)
+        #if (SHADERPASS == SHADERPASS_GBUFFER) && defined(SHADOWS_SHADOWMASK)
```
Comment (seblagarde) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl]: It is not allowed to used this define here: SHADERPASS == SHADERPASS_GBUFFER
This is specific to HDRP and shouldn't be present in shader library

Comment (seblagarde) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl]: UNITY_USE_PROBESOCCLUSION_ARRAY => does this mean that shadowmak  is supported by instancing as well ?

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl [hunk 2]
```diff
@@ -324,10 +322,10 @@
     static float4x4 unity_MatrixITMV_Instanced;
     void UnitySetupCompoundMatrices()
     {
-        unity_MatrixMVP_Instanced = mul(UNITY_MATRIX_VP, unity_ObjectToWorld);
-        unity_MatrixMV_Instanced = mul(UNITY_MATRIX_V, unity_ObjectToWorld);
+        unity_MatrixMVP_Instanced = mul(UNITY_MATRIX_VP, UNITY_MATRIX_M);
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl]: @phi-lira Do you have any used for these matrix ? We don't use it in HD (and we must not use them as we provide our own matrix from C#). I think we should remove this call  (UnitySetupCompoundMatrices and its content

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl (hunk 1 above)
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl]: (SHADERPASS == SHADERPASS_GBUFFER) is not allowed here as we are in shader library and this is HDRP scpecific.

SRP (HD or other) handle shadow mask differently than legacy. There is no distinction between deferred and forward shadow mask for SRP. so this code should only test defined(SHADOWS_SHADOWMASK)

Note: I am wondering is shadow mask instancing is supported ? Seems that there is some work done on it, will ask Yao

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl (hunk 2 above)
Comment (phi-lira) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/UnityInstancing.hlsl]: I don't use them. not even the UNITY_MATRIX_MVP as it's not performance wise.

**Comment by Wadarass** [comment]
@sebastienlagarde Updated with the changes requested.  Just nuked all the compound matrices from UnityInstancing.hlsl and removed Shaderpass reference.

**Pull request merged**